### PR TITLE
#154 refactoring of method and addition of tests

### PR DESCRIPTION
**Pull request by krzyk**
#154

I've refactored `TsDoc.route` slightly (removed anonymous classes where they were not needed) and added tests for routing.

### CODE REVIEW

**Comment by karato** [comment]
Many thanks, I will find someone to review it before we merge

**Comment by karato** [comment]
@darkled do this review

File: src/main/java/com/nerodesk/takes/doc/TsDoc.java [hunk 1]
```diff
@@ -87,37 +84,14 @@ public Take route(final Request req) throws IOException {
...
                 "/doc/eject-friend",
-                new Takes() {
-                    @Override
-                    public Take route(final Request rst) {
-                        return new TkEjectFriend(
-                            doc, href.param("friend").iterator().next()
-                        );
-                    }
-                }
+                new TkEjectFriend(doc, href.param("friend"))
```
Comment (asaen) [inline src/main/java/com/nerodesk/takes/doc/TsDoc.java]: can we avoid repeat of `"friend"` literal here?

**Comment by asaen** [comment]
@krzyk see a minor comment above and resolve merge conflicts please

**Comment by krzyk** [comment]
@darkled after recent refactorings in master most of my changed didn't make sense, please have a look again

**Comment by asaen** [comment]
@rultor merge

File: src/test/java/com/nerodesk/takes/doc/TkDocTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * TkDoc can route to friend addition.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void routesToFriendAddition() throws Exception {
+        MatcherAssert.assertThat(
+            new TkDoc(new MkBase())
+                .act(this.request("/doc/add-friend?file=name", "friend=one")),
+            Matchers.instanceOf(RsForward.class)
```
Comment (yegor256) [inline src/test/java/com/nerodesk/takes/doc/TkDocTest.java:59]: bad idea.. let's wait for this: https://github.com/yegor256/takes/issues/166 and do it properly

**Comment by yegor256** [comment]
@krzyk this hamcrest matching should be done differently, see https://github.com/yegor256/takes/issues/166

**Comment by karato** [comment]
@darkled the project is temporarily on pause, please stop working with this task, until further instructions

**Comment by karato** [comment]
@darkled we're not on pause any more, you may continue to work with the task

**Comment by asaen** [comment]
@karato we're waiting for yegor256/takes#166 here

**Comment by asaen** [comment]
@krzyk yegor256/takes#166 is closed, I guess you can continue

**Comment by krzyk** [comment]
@darkled I think this one doesn't make sense now as the puzzle is no longer there

**Comment by karato** [comment]
@darkled I just added **19 mins** to your account, many thanks for your contribution.. 56729227 spent here... you're getting extra minutes here (c=4)... +19 to your rating, your total score is [+2122](http://www.netbout.com/b/35091?open=rating)

**Pull request closed without merging**